src/Validation/Validator.php

```diff
@@ -525,6 +527,14 @@ protected function _fieldIsEmpty($data)
         if (empty($data) && $data !== '0' && $data !== false && $data !== 0 && $data !== 0.0) {
             return true;
         }
+        $isArray = is_array($data);
+        if ($isArray && (isset($data['year']) || isset($data['hour']))) {
+            $value = implode('', $data);
+            return strlen($value) === 0;
+        }
+        if ($isArray && isset($data['name'], $data['type'], $data['tmp_name'], $data['error'])) {
+            return (int)$data['error'] === UPLOAD_ERR_NO_FILE;
```

Comment:
@jadb The post says "if a user ever does upload a file that exceeds the upload_max_filesize directive in php.ini, the $_FILES superglobal is empty", so if the field isn't populated at all not empty check doesn't come in picture. It's the job of requirePresence to ensure field's existence.